aspnetcore/fundamentals/hosting.md

```diff
@@ -1,225 +1,530 @@
 ---
-title: Hosting in ASP.NET Core| Microsoft Docs
-author: ardalis
-description: Introduction to web hosts in ASP.NET Core.
-keywords: ASP.NET Core, web host, IWebHost
+title: Hosting in ASP.NET Core
+author: guardrex
+description: Learn about the web host in ASP.NET Core, which is responsible for app startup and lifetime management.
+keywords: ASP.NET Core,web host,IWebHost,WebHostBuilder,IHostingEnvironment,IApplicationLifetime
 ms.author: riande
 manager: wpickett
-ms.date: 08/02/2017
+ms.date: 08/31/2017
 ms.topic: article
 ms.assetid: 4e45311d-8d56-46e2-b99d-6f65b648a277
 ms.technology: aspnet
 ms.prod: asp.net-core
 uid: fundamentals/hosting
 ms.custom: H1Hack27Feb2017
 ---
-# Introduction to hosting in ASP.NET Core
+# Hosting in ASP.NET Core
 
-By [Steve Smith](http://ardalis.com)
+By [Luke Latham](https://github.com/guardrex)
 
-To run an ASP.NET Core app, you need to configure and launch a host using `WebHostBuilder`.
+ASP.NET Core apps configure and launch a *host*, which is responsible for app startup and lifetime management. At a minimum, the host configures a server and a request processing pipeline.
 
-## What is a Host?
+## Setting up a host
 
-ASP.NET Core apps require a *host* in which to execute. A host must implement the `IWebHost` interface, which exposes collections of features and services, and a `Start` method. The host is typically created using an instance of a `WebHostBuilder`, which builds and returns a  `WebHost` instance. The `WebHost` references the server that will handle requests. Learn more about [servers](servers/index.md).
-
-### What is the difference between a host and a server?
-
-The host is responsible for application startup and lifetime management. The server is responsible for accepting HTTP requests. Part of the host's responsibility includes ensuring the application's services and the server are available and properly configured. You can think of the host as being a wrapper around the server. The host is configured to use a particular server; the server is unaware of its host.
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-## Setting up a Host
+Create a host using an instance of [IWebHost](/dotnet/api/microsoft.aspnetcore.hosting.iwebhost). This is typically performed in your app's entry point, the `Main` method. In the project templates, `Main` is located in *Program.cs*. A typical *Program.cs* calls [CreateDefaultBuilder](/dotnet/api/microsoft.aspnetcore.webhost.createdefaultbuilder) to build the host:
```

Comment:
Create a host using an instance of WebHostBuilder

A typical *Program.cs* calls [CreateDefaultBuilder](/dotnet/api/microsoft.aspnetcore.webhost.createdefaultbuilder) to start setting up a host: